### PR TITLE
Fixed a race condition that could cause some tests not being added correctly to a test run

### PR DESCRIPTION
**Pull request by Andarist**
`CreateTestRunShard` is an async call and it's not guaranteed that it completes before we start working on `AddTestsToShard` (or even `CompleteTestRunShard`).

This fixes that problem by always awaiting the shard id "task" before we continue with other calls that depend on this shard ID.

### CODE REVIEW

**Review by hbenl — APPROVED** [review]

File: packages/test-utils/src/reporter.ts [hunk 1]
```diff
@@ -200,6 +202,7 @@ class ReplayReporter<TRecordingMetadata extends UnstructuredMetadata = Unstructu
   upload = false;
   filter?: (r: RecordingEntry<TRecordingMetadata>) => boolean;
   recordingsToUpload: ExternalRecordingEntry<TRecordingMetadata>[] = [];
+  _testRunShardIdPromise: Promise<TestRunPendingWork> | null = null;
```
Comment (bvaughn) [inline packages/test-utils/src/reporter.ts:205]: Any reason not to just make this private?
```suggestion
  private testRunShardIdPromise: Promise<TestRunPendingWork> | null = null;
```

Comment (Andarist) [reply packages/test-utils/src/reporter.ts:205]: Not making it private is an oversight on my part but I think it's still worth maintaining the `_` convention, even when using `private` keyword.

Comment (bvaughn) [reply packages/test-utils/src/reporter.ts:207]: Huh. Why?

Comment (Andarist) [reply packages/test-utils/src/reporter.ts:207]: `private` doesn't hide it at a runtime level, just at the type level. I find it useful to know what's private and what's not when inspect values in the debugger

Comment (bvaughn) [reply packages/test-utils/src/reporter.ts]: That's fair. I guess it depends on what our compile target is. I was assuming TS would get compiled to `#testRunShardIdPromise` but I guess it doesn't?

Comment (Andarist) [reply packages/test-utils/src/reporter.ts]: Ye, it doesn't do that - regardless of the target. To use private fields u have to use `#` syntax, not the `private` keyword

Comment (bvaughn) [reply packages/test-utils/src/reporter.ts]: TIL

Bad assumption on my part about TS :) Thanks for sharing

**Pull request merged**